File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys
input = sys.stdin.readline

n =  int(input().strip())
#print(n)
arr = sorted([list(map(int, input().split())) for _ in range(n)], key=lambda x : (x[1],x[0]))
res = []
'''
1. 회의가 빠르게 끝나면 더 많은 회의 가능
2. 회의가 끝나는 시간과 가까운 시작 시간
'''
previous_end_time = 0
for i in arr:
    start, end = i[0],i[1]
    if start>=previous_end_time:
        previous_end_time=end
        res.append(i)
print(len(res))

File: 백준/Gold/1931. 회의실 배정/회의실 배정.js
const readline = require("readline")
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

const lines = []
rl.on("line", line => lines.push(line)).on("close", () => {

    const N = +lines[0]
    const arr = lines.slice(1).map((item)=>item.split(" ").map(Number))
    arr.sort((a,b)=>(a[1]-b[1]||a[0]-b[0]))
    let answer = 0
    let end = -1
    for(let e of arr){
        if(e[0]>=end){
            answer++
            end = e[1]
        }
    }
    
    // console.log(arr)
    console.log(answer)

})